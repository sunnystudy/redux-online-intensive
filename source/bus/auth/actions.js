//Types
import { types } from './types';

export const authActions = {
    //Sync
    authenticate: () => {
        return {
            type: types.AUTHENTICATE,
        };
    },
    initialize: () => {
        return {
            type: types.INITIALIZE,
        };
    },
    logout: () => {
        return {
            type: types.LOGOUT,
        };
    },
    //Async
    authenticateAsync: () => {
        return {
            type: types.AUTHENTICATE_ASYNC,
        };
    },
    loginAsync: (credentials) => {
        return {
            type:    types.LOGIN_ASYNC,
            payload: credentials,
        };
    },
    signupAsync: (userData) => {
        return {
            type:    types.SIGNUP_ASYNC,
            payload: userData,
        };
    },
    initializeAsync: () => {
        return {
            type: types.INITIALIZE_ASYNC,
        };
    },
    logoutAsync: () => {
        return {
            type: types.LOGOUT_ASYNC,
        };
    },
};
