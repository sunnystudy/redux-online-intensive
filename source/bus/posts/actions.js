//Types
import { types } from "./types";

export const postActions = {
    fillPosts: (posts) => {
        return {
            type:    types.FILL_POSTS,
            payload: posts,
        };
    },
    fetchPostsAsync: () => {
        return {
            type: types.FETCH_POSTS_ASYNC,
        };
    },
    createPost: (post) => {
        return {
            type:    types.CREATE_POST,
            payload: post,
        };
    },
    createPostAsync: (comment) => {
        return {
            type:    types.CREATE_POST_ASYNC,
            payload: comment,
        };
    },
};
