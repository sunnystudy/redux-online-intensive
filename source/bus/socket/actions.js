//Actions
import { socket } from '../../init/socket';
import { uiActions } from '../ui/actions';
import { postActions } from '../posts/actions';

export const socketActions = {
    listenConnection: () => (dispatch) => {
        socket.on('connect', () => {
            dispatch(uiActions.setOnlineState());
        });
        socket.on('disconnect', () => {
            dispatch(uiActions.setOfflineState());
        });
    },
    listenPosts: () => (dispatch, getState) => {
        socket.on('create', (event) => {
            const { data: post } = JSON.parse(event);

            dispatch(postActions.createPost(post));
        });
        socket.on('remove', (event) => {
            const { data: postId } = JSON.parse(event);

            dispatch(postActions.removePost(postId));
        });
        socket.on('like', (event) => {
            const { data, meta } = JSON.parse(event);

            if (meta.action === 'like') {
                const liker = getState()
                    .users.find((user) => user.get('id') === data.userId)
                    .delete('avatar');

                dispatch(postActions.likePost({
                    postId: data.postId,
                    liker,
                }));
            } else {
                dispatch(postActions.unlikePost(data));
            }
        });
    },
};
