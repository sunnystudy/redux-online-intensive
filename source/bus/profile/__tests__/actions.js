//Actions
import { profileActions } from "../actions";

describe('profile actions: ', () => {
    test('fillProfile', () => {
        expect(profileActions.fillProfile(__.userProfile)).toMatchSnapshot();
    });
    test('updateAvatar', () => {
        expect(profileActions.updateAvatar(__.url)).toMatchSnapshot();
    });
    test('clearProfile', () => {
        expect(profileActions.clearProfile()).toMatchSnapshot();
    });
});
