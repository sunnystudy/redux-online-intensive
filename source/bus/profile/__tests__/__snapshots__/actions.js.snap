// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions:  clearProfile 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`profile actions:  fillProfile 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`profile actions:  updateAvatar 1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;
